WordPress/Classes/ViewRelated/Blog/Settings/SiteSettingsView.swift

```diff
@@ -0,0 +1,137 @@
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
+                if presentationMode.wrappedValue.isPresented {
+                    closeButton
+                }
+            }
+        }
+        .onReceive(viewModel.onDismissableError) {
+            SVProgressHUD.showDismissibleError(withStatus: $0)
+        }
+        .onAppear {
+            Task { await viewModel.refresh() }
+        }
+    }
+
+    @ViewBuilder
+    private var sections: some View {
+        Section(header: Text(Strings.Sections.general)) {
+            siteTitleRow
+        }
+    }
+
+    // MARK: - General
+
+    private var siteTitleRow: some View {
+        withAdminNavigationLink(destination: {
+            SettingsTextEditView(
+                value: settings.name,
+                placeholder: Strings.General.siteTitlePlaceholder,
+                onCommit: viewModel.updateSiteTitle
+            )
+            .navigationTitle(Strings.General.siteTitle)
+        }) {
+            SettingsCell(title: Strings.General.siteTitle, value: settings.name ?? Strings.General.siteTitlePlaceholder)
+        }
+    }
+
+    // MARK: - Helpers
+
+    @ViewBuilder
+    private func withAdminNavigationLink<T: View, U: View>(
+        @ViewBuilder destination: () -> T,
+        @ViewBuilder content: () -> U
+    ) -> some View {
+        if blog.isAdmin {
+            NavigationLink(destination: destination(), label: content)
+        } else {
+            content()
+        }
+    }
+
+    private var closeButton: some View {
+        Button(action: { presentationMode.wrappedValue.dismiss() }) {
+            Text(Strings.done)
+                .font(.body.weight(.medium))
+                .foregroundColor(Color.primary)
+        }
+    }
+}
+
+private struct SettingsCell: View {
+    let title: String
+    let value: String
+
+    var body: some View {
+        HStack {
+            Text(title)
+                .layoutPriority(1)
+            Spacer()
+            Text(value)
+                .foregroundColor(.secondary)
+        }
+        .lineLimit(1)
+    }
+}
+
+private struct SettingsTextEditView: UIViewControllerRepresentable {
+    let value: String?
+    let placeholder: String
+    var hint: String?
+    let onCommit: ((String)) -> Void
+
+    func makeUIViewController(context: Context) -> SettingsTextViewController {
+        let viewController = SettingsTextViewController(text: value ?? "", placeholder: placeholder, hint: hint ?? "")
+        viewController.onValueChanged = onCommit
+        return viewController
+    }
+
+    func updateUIViewController(_ uiViewController: SettingsTextViewController, context: Context) {
+        // Do nothing
+    }
+}
+
+private extension SiteSettingsView {
+    enum Strings {
+        static let title = NSLocalizedString("Settings", comment: "Title for screen that allows configuration of your blog/site settings.")
```

Comment:
Could you please use reverse DNS keys for localized strings? We lack a way to enforce them at the moment, but the rationale is that they help us disambiguate duplicated strings used in different contexts. More at #19028

Reply:
Yes, that is a good point. I was reluctant to do that in the same PR because these are the existing keys from settings, and changing them will require updating all the localizable strings files. But it shouldn't take much time and should not create merge conflicts, so I'll update the keys.

I'll have to be careful in case other screens use the same keys since they are not namespaced.

Reply:
> these are the existing keys from settings, and changing them will require updating all the localizable strings files (presumable we want to keep the localizations?).

So I'm guessing these are duplicated here for usage convenience in SwiftUI? E.g. this is the "original" title localized string:

https://github.com/wordpress-mobile/WordPress-iOS/blob/e4db21b30a9fa50809d26493b670e7b39b9fabbe/WordPress/Classes/ViewRelated/Blog/Site%20Settings/SiteSettingsViewController.m#L124

For what is worth, localizations are cheap to change, and it wouldn't be a problem, IMHO, if this added more strings to localize.

For context, we generate new `.strings` via custom `genstrings` wrappers during code freeze. These are sent to GlotPress for translations, and every time we build a binary for deployment we download up-to-date strings.

> You also have to be careful in case other screens use the same keys since they are not namespaced.

Exactly. Updating to use a reverse-DNS key will create a unique<sup>*</sup> localized string, e.g.:

```suggestion
        static let title = NSLocalizedString("sitesettings.title", value: "Settings", comment: "Title for screen that allows configuration of your blog/site settings.")
```

Whereas currently the "Settings" key+value is shared by four instances:

https://github.com/wordpress-mobile/WordPress-iOS/blob/e4db21b30a9fa50809d26493b670e7b39b9fabbe/WordPress/Resources/en.lproj/Localizable.strings#L7725-L7730

---

`*` - `genstrings` is capable of detecting duplicated key usages, but doesn't fail when that happens. We have it in the roadmap to at some point address that.

Reply:
There aren't going to be too many new strings, so I'll add new ones, as you suggest. Updated: https://github.com/wordpress-mobile/WordPress-aiOS/pull/20838/commits/b16704dbc6fc41516b9bc71a2b39c3ef6983ad41.